### ISSUE TITLE
Found unparsable section: ' notnull'

### ISSUE THREAD

**Issue by nolanbconaway** [alanmcruickshank/sqlfluff#426]
This query is definitely valid postgres:

```sql
select * from t where c notnull
```

Alas, fluff disagrees!

```sh
$ echo 'select * from t where c notnull' | sqlfluff lint - --dialect postgres
== [stdin] FAIL
L:   1 | P:  24 |  PRS | Found unparsable section: ' notnull'
```

For the record:

```sh
$ sqlfluff version    
0.3.5
```

**Comment by alanmcruickshank** [comment]
Interesting! I wasn't familiar with that syntax.

Yes it is valid in postgres (https://www.postgresql.org/docs/9.2/sql-syntax-lexical.html#SQL-SYNTAX-OPERATORS), but not in mysql (https://dev.mysql.com/doc/refman/5.7/en/non-typed-operators.html) or snowflake (https://docs.snowflake.com/en/sql-reference/operators-logical.html). Interestingly isn't also not supported in Redshift (https://docs.aws.amazon.com/redshift/latest/dg/r_null_condition.html) either which means this might be the first feature which justifies a new redshift dialect!

Postgres classifies it as an operator, although I think sqlfluff should probably treat `NOTNULL` and `ISNULL` as keywords for our purposes. The logic for this should go in `Expression_A_Grammar`, but that expression is already quite big a probably needs a little surgery to make it a bit more configurable.

@nolanbconway, want to have a go at this? I don't immediately have a way of doing this tidily, but I'm open to suggestions!
